codec-http2/src/main/java/io/netty/handler/codec/http2/Http2FrameCodec.java

```diff
@@ -146,11 +227,49 @@ public Http2FrameCodec(boolean server, Http2FrameLogger frameLogger) {
         decoder.frameListener(new FrameListener());
         http2Handler = new InternalHttp2ConnectionHandler(decoder, encoder, initialSettings);
         http2Handler.connection().addListener(new ConnectionListener());
+        streamKey = connection().newKey();
         this.server = server;
     }
 
-    Http2ConnectionHandler connectionHandler() {
-        return http2Handler;
+    // TODO(buchgr): Discuss: Should this method be thread safe?
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1135)